src/libraries/Microsoft.PowerFx.Repl/Repl.cs

```diff
@@ -403,6 +370,23 @@ await this.Output.WriteLineAsync($"Error: Can't set '{name}' to a Void value.",
 
             // Show parse/bind errors 
             var errors = check.ApplyErrors();
+
+            if (!check.IsSuccess)
+            {
+                // Checking for named formulas and/or udfs.
+                try
+                {
+                    this.Engine.AddUserDefinitions(expression, parseCulture: this.ParserOptions.Culture, symbolTable: currentSymbolTable, onUpdate: OnFormulaUpdate);
+
+                    return new ReplResult();
+                }
+                catch (InvalidOperationException ex)
+                {
```

Comment:
This isn't ideal - we should be able to reliably check for parse errors. See comment above.